Makefile.dep

```diff
@@ -69,6 +69,11 @@ ifneq (,$(filter gnrc_mac,$(USEMODULE)))
   endif
 endif
 
+# Don't register netdevs if there is only a single one  of them
+ifeq (,$(filter gnrc_netif_single,$(USEMODULE)))
+  USEMODULE += netdev_register
+endif
```

Comment:
This is pulling `netdev_register` almost always now. Although it is a pseudomodule I think the dependency should be changed. AFIK there is no equivalent to `gnrc_netif_single` for other stacks, right? (@miri64 ?) Should this be added when some `netdev_%` module is used instead?

Reply:
This is affecting https://github.com/RIOT-OS/RIOT/pull/13565 now, as the unneeded dependency `netdev_register` is pulled.

Reply:
`netdev_register` is a pseudomodule, so it won't do anything on it's own.
If not `netdev_` code is used, this will not pull in any additional code.

Reply:
But it changes the dependency resolution. This can indeed change the binary,

Reply:
Isn't `netdev_register` a sub-module to `netdev`? If not, IMHO there is nothing speaking against it being one.

Reply:
I guess we just missed a step in this discussion: https://github.com/RIOT-OS/RIOT/pull/13746#discussion_r461458518

Reply:
how to proceed with that?

Reply:
but you just provided the solution yourself https://github.com/RIOT-OS/RIOT/pull/13565#discussion_r497556842

Reply:
I thought it was not intended to have a reverse dependency to `gnrc_netif_single`

Reply:
I am fine with a reverse dependency how you implemented it. My original concern was about GNRC code in the network device code.